lib/google-apis/auth-client.js

```diff
@@ -69,6 +81,83 @@ function AuthClient(projectId, shouldReportErrors, key) {
 }
 
 /**
+ * Adds the given callback as a listener to the `init` event which will be
+ * broadcast once the client has inited the underlying transport and generated
+ * or failed to generate a token.
+ * @function addInitListener
+ * @param {Function} callback - the listener to callback on when the client has
+ *  inited
+ * @returns {Undefined} - does not return anything
+ */
+AuthClient.prototype.addInitListener = function(callback) {
+
```

Comment:
Upon further reading, I noticed that this is public API for AuthClient. In that case, could you subclass AuthClient from EventEmitter, and let the users call `.on`, `.emit` directly?